### ISSUE TITLE
libressl: netcat doesn't work on macos

### ISSUE THREAD

**Issue by sdier** [NixOS/nixpkgs#101601]
**Describe the bug**
netcat from libressl doesn't work on macos

**To Reproduce**
1. as root: dtruss -W nc
2. nc localhost 22
3. note that the socket open shown in dtruss shows error 41

**Expected behavior**
Connects.

**Additional context**
I have opened https://github.com/libressl-portable/portable/issues/631

I don't think anyone actually ever used this successfully on macos.  I suspect it should be disabled from building for this platform if possible until its fixed so nobody else has to go through the debugging.

**Notify maintainers**
@fpletz 
@thoughtpolice 

**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS `
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(sdier): `"home-manager, nixpkgs-21.03pre246624.cfed29bfcb2"`
 - nixpkgs: `/Users/sdier/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: libressl.nc
```

**Comment by sdier** [comment]
"I don't think anyone actually ever used this successfully on macos"

Actually I do not know if error 41 was always returned for this combination of options.

**Comment by risicle** [comment]
A painful one.

If we were to disable this for macos, might we have another means of providing macos users with a bsd-style netcat?

**Comment by risicle** [comment]
The other route for BSD netcat on macos is of course https://opensource.apple.com/tarballs/netcat/, but nobody has yet build an `apple-source-releases` derivation for it.

**Comment by risicle** [comment]
Oh and I tried a naive switch of the `SOCK_CLOEXEC`/`SOCK_NONBLOCK` bits the other day (up to.. `0x4000000` and `0x2000000` i think?) and it didn't seem to work.

**Comment by Gabriel439** [comment]
If somebody running across this issue needs a work-around: you can use `pkgs.netcat-gnu` instead of `pkgs.netcat`.  `pkgs.netcat-gnu` works on macOS

**Comment by risicle** [comment]
The issue is that `netcat-gnu` isn't _quite_ the same. It doesn't support e.g. SOCKS5 in the same way.

**Comment by sebastianblunt** [comment]
This should be fixed by the update to libressl 3.5.2.
